src/view/RoomsView.java

```diff
@@ -94,38 +150,73 @@ public void keyReleased(KeyEvent e) {}
         evt -> {
           if (evt.getSource().equals(refreshButton)) {
             RoomsState currentState = viewModel.getState();
-            String roomUid = currentState.getRoomUid();
             User user = currentState.getUser();
+            String roomUid = currentState.getRoomUid();
 
             for (var room : currentState.getAvailableRooms()) {
               if (room.getUid().equals(roomUid)) {
                 roomsController.loadMessages(room, user);
-                break;
               }
             }
           }
         });
+
+    addUserButton.addActionListener(
+        evt -> {
+          if (evt.getSource().equals(addUserButton)) {
+            RoomsState currentState = viewModel.getState();
+            String email = currentState.getEmail();
+            String roomUid = currentState.getRoomUid();
+            if (email != null) {
+              User user = currentState.getUser();
+              for (var room : currentState.getAvailableRooms()) {
```

Comment:
this logic should probably be in the interactor with a method called getRoomByUid

Reply:
I actually had this written down on paper. Guess I forgot lol.

Reply:
It shouldn't be in the interactor, should be in `RoomsState`.